mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -376,12 +376,24 @@ pub async fn create_wallet_db(
         move || -> Result<WalletDb<ZcoinConsensusParams>, MmError<ZcoinClientInitError>> {
             let db = WalletDb::for_path(wallet_db_path, consensus_params)
                 .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
+            let extrema = db.block_height_extrema()?;
+            let min_sync_height = extrema.map(|(min, _)| u32::from(min));
+            let init_block_height = checkpoint_block.clone().map(|block| block.height);
+
             run_optimization_pragmas(db.sql_conn())
                 .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
             init_wallet_db(&db).map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
-            if db.get_extended_full_viewing_keys()?.is_empty() {
-                init_accounts_table(&db, &[evk])?;
-                if let Some(block) = checkpoint_block {
+
+            // Check if the initial block height is less than the previous synchronization height,
+            // Rewind the walletdb to the minimum possible height.
+            // and if a checkpoint block is available, intitialize the walletdb with this block.
+            if db.get_extended_full_viewing_keys()?.is_empty() || init_block_height != min_sync_height {
+                info!("Older/Newer sync height detected!, rewinding walletdb to new height: {init_block_height:?}");
+                let mut wallet_ops = db.get_update_ops().expect("get_update_ops always returns Ok");
+                wallet_ops
+                    .rewind_to_height(u32::MIN.into())
+                    .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
```

Comment:
Can these 2 conditions be seperated? e.g.
```rust
// Check if the initial block height is less than the previous synchronization height,
// Rewind the walletdb to the minimum possible height.
if init_block_height != min_sync_height {
    info!("Older/Newer sync height detected!, rewinding walletdb to new height: {init_block_height:?}");
    let mut wallet_ops = db.get_update_ops().expect("get_update_ops always returns Ok");
    wallet_ops
        .rewind_to_height(u32::MIN.into())
        .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
}

if db.get_extended_full_viewing_keys()?.is_empty() {
```

Reply:
nop, this doesn't work because `get_extended_full_viewing_keys()` won't still be empty after this, so there won't be any way to initialise `walletdb` with new block
```rust
if init_block_height != min_sync_height {
    info!("Older/Newer sync height detected!, rewinding walletdb to new height: {init_block_height:?}");
    let mut wallet_ops = db.get_update_ops().expect("get_update_ops always returns Ok");
    wallet_ops
        .rewind_to_height(u32::MIN.into())
        .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
}
```

Reply:
Thanks. We will leave it as is then.